### DISCUSSION TITLE
Changes to {Matrix} ABI could cause incompatibility with TMB version

### DISCUSSION THREAD

**Discussion by ChristineStawitz-NOAA** [NOAA-FIMS/FIMS#1242] [Developers]
### Issue details

Once we start using sparse matrices we could run into versioning issues between Matrix and TMB re:this thread: https://groups.google.com/g/tmb-users/c/-GhmuuDP_OQ

We can probably wait a bit to address but should implement a solution sometime in M2
 
Related issue in ghactions4r:  https://github.com/nmfs-fish-tools/ghactions4r/issues/106

**Comment by msupernaw** [comment]
Based on the forums, it looks like this is an issue with the prototypes in the SuiteSparse headers not matching the ABI. This will likely be resolved soon and I can't see any impact on FIMS. I think we can change the label to "needs more info" for now.

**Comment by kellijohnson-NOAA** [comment]
@Andrea-Havron-NOAA I see that you commented on the TMB thread regarding this issue. Is this still something that we should be thinking about or is it no longer relevant?

**Comment by Andrea-Havron-NOAA** [comment]
> [@Andrea-Havron-NOAA](https://github.com/Andrea-Havron-NOAA) I see that you commented on the TMB thread regarding this issue. Is this still something that we should be thinking about or is it no longer relevant?

@kellijohnson-NOAA, I am not sure if this is still relevant. We may need to implement something similar to [glmmTMB](https://github.com/glmmTMB/glmmTMB/blob/8e4a3f35fb3a74f56b27465ac2560594ce0e6824/glmmTMB/R/utils.R#L224)

**Comment by kellijohnson-NOAA** [comment]
I think that I will leave this open but I am going to move it to a Discussion because there are 0 actionable tasks for this right now. We can always require Matrix be greater than version 1.3.0 so we do not have to have the if statement that glmmTMB uses. Right now Matrix is on version 1.7.4 but google also notes that version 1.7.0, released in 2024, incremented the Matrix ABI (Application Binary Interface) version from 1 to 2, which required packages linking to it to be rebuilt from source.
